src/core/gcc.rs

```diff
@@ -277,6 +277,12 @@ pub fn server_security_data() -> Component {
 
 /// Actually we have no more classic channel
 pub fn channel_def(name: &String, options: u32) -> Component {
+    // Channel name is exactly 8 null-terminated ANSI characters.
+    let name = if name.len() >= 8 {
+        (name[0..8]).to_string()
+    } else {
+        name.clone() + &"\x00".repeat(8 - name.len())
+    };
     component![
         "name"=> name.as_bytes().to_vec(),
         "options" => U32::LE(options)
```

Comment:
Nit: Something like this would be more idiomatic rust:

```rust
pub fn channel_def(mut name: &str, options: u32) -> Component {
    if name.len() > 8 {
        name = &name[..8];
    }
    component![
        "name"=> format!("{:\x00<8}",name).into_bytes(),
        "options" => U32::LE(options),
    ]
}
```

(uses `&str` instead of `&String`, and eliminates a number of intermediate allocations by shrinking in-place, using the `format!` macro for fill, and avoiding ownership in intermediate values)